src/CommunityToolkit.Maui.SourceGenerators.Internal/Helpers/AttributeExtensions.cs

```diff
@@ -22,6 +21,35 @@ public static string GetNamedTypeArgumentsAttributeValueByNameAsCastedString(thi
 			return data.Value is null ? placeholder : $"({data.Type}){data.Value.ToString().ToLowerInvariant()}";
 		}
 
+		if (data.Kind is TypedConstantKind.Enum && data.Type is not null && data.Value is not null)
+		{
+			var members = data.Type.GetMembers();
+			return $"({data.Type}){members[(int)data.Value]}";
```

Comment:
maybe this is missed from previous iterations, but this can break, since enums can inherit from other types, like byte, long, etc.

[Here's a sample on how it can break](https://sharplab.io/#v2:C4LgTgrgdgPgAgJgAwFgBQiCM710wTgAoAbAeygHMA6AWQEMAPANTuIgFMBKAblwMICWUYLUYs2XXmj5EeMwoOGcAKuwDOwAKJQIAWyoBlUrvYAJCBXbiOc6XZxp2O3QAJVG7XpcgXZSugBvdBcQlwARdgAzOghiYAAaYNCjE3NLa3YXAF4XBEwAFgB2fIAOAGYANlKXAHoalyERemZWDhcAahdsNABfIA==)

Reply:
Your example demonstrates casting an `enum: long` to an `int`. This code doesn't do that. 

Here's an example of the property generated for `enum Status : long`:

```cs
public static readonly global::Microsoft.Maui.Controls.BindableProperty InvoiceStatusProperty = global::Microsoft.Maui.Controls.BindableProperty.Create("InvoiceStatus", typeof(TestNamespace.Status), typeof(TestNamespace.TestClass), null, Microsoft.Maui.Controls.BindingMode.OneWay, null, null, null, null, null);

public partial TestNamespace.Status InvoiceStatus { get => (TestNamespace.Status)GetValue(InvoiceStatusProperty); set => SetValue(InvoiceStatusProperty, value); }

Reply:
Ah - got it! My unit test was missing a `DefaultValue` and it needed to be greater than `int.MaxValue` to trigger the bug. 

Fixed it! Fantastic catch - thanks Pedro!!!

I've updated the source generator to use the enum's value, casting it back it its enum type. For example `Status.Rejected` (below), now generates  `(TestNamespace.Status)9223372036854775807` when used as the `DefaultValue`.

```cs
public enum Status : long
{
    Pending = 0,
    Approved = 1,
    Rejected = long.MaxValue
}
```